test/hashing.jl

```diff
@@ -302,3 +302,5 @@ struct AUnionParam{T<:Union{Nothing,Float32,Float64}} end
     # test hashing of rational with odd denominator
     @test hash(5//3) == hash(big(5)//3)
 end
+
+@test Base.infer_effects(hash, Tuple{Type{Int}, UInt}) == Core.Compiler.EFFECTS_TOTAL
```

Comment:
```suggestion
@test Core.Compiler.is_foldable_nothrow(Base.infer_effects(hash, Tuple{Type{Int}, UInt}))
```
is the way to test this.

Reply:
Thank you!